src/components/ui/calendar.tsx

```diff
@@ -1,15 +1,30 @@
 import { ChevronLeft, ChevronRight } from "lucide-react";
 import { DayPicker, DayPickerProps } from "react-day-picker";
-
 import { cn } from "@/lib/utils";
 import { buttonVariants } from "@/components/ui/button";
+import { useEffect, useState } from "react";
+import { fetchGoogleCalendarEvents } from "@/calendar-api";
 
 export type CalendarProps = DayPickerProps & {
 	className?: string;
 	classNames?: Partial<DayPickerProps["classNames"]>;
 };
 
 function Calendar({ className, classNames, showOutsideDays = true, ...props }: CalendarProps) {
+	// Unavailable dates conditional formatting prep functions
+	const [unavailableDates, setUnavailableDates] = useState<string[]>([]);
+
+	useEffect(() => {
+		const fetchDates = async () => {
+			const dates = await fetchGoogleCalendarEvents();
+			setUnavailableDates(dates);
+		};
+		fetchDates();
+	}, []);
```

Comment:
```suggestion
	useEffect(() => {
		const fetchDates = async () => {
			const dates = await fetchGoogleCalendarEvents();
			setUnavailableDates(dates);
		};
		void fetchDates();
	}, []);
```

This is a small thing, but async functions should be either awaited or voided. It helps the javascript garbage collector to manage memory a little better, while keeping the same functionality you have now.